### PR TITLE
fix duplicate custom trait sending| EPD-2498

### PR DESCRIPTION
**Pull request by hankim813**
This fixes issues #34.
This PR basically cherry picks PR #35 
TL:DR;

- long ago when we had v1 & v2 users, we shipped a fix that removes top level semantic fields from the `custom_attributes` property when we send `.identify()` or `.group()` (user/company) data. 
- we shipped the fix only for v2 (since our plan at the time was to migrate everyone to v2)
- now that intercom asked us to remigrated everyone back to v1, users saw this issue come up
- this PR applies the same fix to v1. 
- this also covers the edge case (which v1 nor v2 caught before) where if you send `traits.company` as a string, you would not be able to set semantic fields like `plan` or `monthly_spend`.

@tsholmes @anoonan @ccnixon @Peripheral1994 @ladanazita 

E2E confirmed.

### CODE REVIEW

**Comment by ccnixon** [comment]
@hankim813 lgtm
